### PR TITLE
[ADD] new module pos_keep_draft_orders to let orders in 'draft' state

### PR DESCRIPTION
**Pull request by legalsylvain**
This module allow users to let a PoS Order in a 'draft' state when closing the session.
Please see full description in OpenERP file.

Remarks are welcome.

Code Coverage of this module : 100%.

Kind regards.

### CODE REVIEW

File: pos_keep_draft_orders/tests/test_pos_keep_draft_orders.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+        # Make partial payment
+        pmp_id = self.pmp_obj.create(cr, uid, {
+            'journal_id': self.cash_journal_id,
+            'amount': 1,
+        })
+        self.pmp_obj.check(cr, uid, [pmp_id], {'active_id': po_id})
+
+        # Try Close Session
+        try:
```
Comment (lmignon) [inline pos_keep_draft_orders/tests/test_pos_keep_draft_orders.py]:
``` python
with self.assertRaises(SomeException) as cm:
    do_something()
```

https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises

Comment (legalsylvain) [inline pos_keep_draft_orders/tests/test_pos_keep_draft_orders.py]: Ow. thanks ! I didn't know this function.

File: pos_keep_draft_orders/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Copyright, Authors and Licence:
+-------------------------------
+    * Copyright:
+        * 2013-Today, GRAP: Groupement Régional Alimentaire de Proximité;
+    * Author:
+        * Julien WESTE;
+        * Sylvain LE GAL (https://twitter.com/legalsylvain);
+    * Licence: AGPL-3 (http://www.gnu.org/licenses/);""",
+    'author': 'GRAP',
```
Comment (lmignon) [inline pos_keep_draft_orders/__openerp__.py]: Plz add OCA as author of OCA addons
In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.(ex: https://github.com/OCA/partner-contact/blob/de1b5b2f16f951020bb89459ddd63dad1cb458d0/__unported__/account_partner_merge/__openerp__.py#L25)

File: pos_keep_draft_orders/model/pos_order.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+class pos_order(Model):
+    _inherit = 'pos.order'
+
+    # Functional Field Section
+    def _get_is_partial_paid(
+            self, cr, uid, ids, name, arg, context=None):
+        res = {}
+        for po in self.browse(cr, uid, ids, context=context):
+            res[po.id] = (po.state == 'draft') and len(po.statement_ids) != 0
```
Comment (lmignon) [inline pos_keep_draft_orders/model/pos_order.py:36]: `!= 0` is not required

``` python
res[po.id] = (po.state == 'draft') and len(po.statement_ids) 
```

File: pos_keep_draft_orders/tests/test_pos_keep_draft_orders.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        super(TestPosKeepDraftOrders, self).setUp()
+        cr, uid = self.cr, self.uid
+
+        self.imd_obj = self.registry('ir.model.data')
+        self.ps_obj = self.registry('pos.session')
+        self.po_obj = self.registry('pos.order')
+        self.pmp_obj = self.registry('pos.make.payment')
+        self.wf_service = netsvc.LocalService('workflow')
+
+        self.pos_config_id = self.imd_obj.get_object_reference(
```
Comment (lmignon) [inline pos_keep_draft_orders/tests/test_pos_keep_draft_orders.py]: `TransactionCase` provides shotcuts to get value from xml_id https://github.com/odoo/odoo/blob/7.0/openerp/tests/common.py#L62

``` python
self.pos_config_id = self.ref('point_of_sale.pos_config_main')
```

Comment (legalsylvain) [inline pos_keep_draft_orders/tests/test_pos_keep_draft_orders.py]: Thanks again.

File: pos_keep_draft_orders/model/pos_order.py (hunk 3 above)
Comment (legalsylvain) [inline pos_keep_draft_orders/model/pos_order.py:36]: In fact yes. It doesn't work without.

**Comment by legalsylvain** [comment]
@lmignon : done.

**Comment by legalsylvain** [comment]
@lmignon. Is it now ok for you ?

Regards.

**Comment by legalsylvain** [comment]
I close this PR after a lot of months without any review.
Feel free to reopen it if you're interested.

**Pull request closed without merging**